### PR TITLE
Update golang to 1.22.2 && update region name on non-DC system

### PR DESCRIPTION
**Pull request by yjian118**
Test passed:
1 initial deployment of AIOSX subcloud with the updated helm chart and image
2 initial deployment of AIOSX subcloud with old helm chart and image, update it with the new chart and image, verify all resources are still reconciled and insync, verified no errors produced in the DM log
3 increase mtu of an interface after the test case2, verified the reconfiguration can trigger a system config update strategy to update the configuration as expected
4 verified the region name is updated on both subcloud and non-DC aiosx, but not updated in system controllers

### CODE REVIEW

**Review by sriram-gn — APPROVED** [review]
LGTM

**Comment by sselvara1** [comment]
Are we going to squash the commits?
Description in this page captures only the test cases, would be good to have description as well. Thanks.

**Comment by yjian118** [comment]
> Are we going to squash the commits? Description in this page captures only the test cases, would be good to have description as well. Thanks.

We can rebase and merge to preserve the commit msgs

File: config/crd/bases/starlingx.windriver.com_hostprofiles.yaml [hunk 1]
```diff
@@ -22,31 +22,38 @@ spec:
...
             description: HostProfileSpec defines the desired state of HostProfile
             properties:
               addresses:
-                description: Addresses defines the list of addresses to be configured
-                  against this host.  Addresses are specific to a single host therefore
-                  they should only be specified if this profile is only going to be
-                  used to configure a single host.
+                description: |-
+                  Addresses defines the list of addresses to be configured against this
+                  host.  Addresses are specific to a single host therefore they should only
```
Comment (hugonicodemos) [inline config/crd/bases/starlingx.windriver.com_hostprofiles.yaml:53]: nit: extra space

File: config/crd/bases/starlingx.windriver.com_hostprofiles.yaml [hunk 2]
```diff
@@ -22,31 +22,38 @@ spec:
...
+                  be specified if this profile is only going to be used to configure a
+                  single
+                  host.
```
Comment (hugonicodemos) [inline config/crd/bases/starlingx.windriver.com_hostprofiles.yaml:56]: nit: extra line

File: config/crd/bases/starlingx.windriver.com_hostprofiles.yaml [hunk 3]
```diff
@@ -858,12 +956,13 @@ spec:
...
-                          description: Journal defines another OSD device to be used
-                            as the journal for this OSD device.
+                          description: |-
+                            Journal defines another OSD device to be used as the journal for this
+                            OSD device.
                           properties:
                             location:
                               description: "Location defines\tthe OSD device path
-                                to be used as the Journal OSD for this logical device."
+                                to be used as the Journal OSD for\nthis logical device."
```
Comment (hugonicodemos) [inline config/crd/bases/starlingx.windriver.com_hostprofiles.yaml:965]: do we need the \n ?

**Review by hugonicodemos — CHANGES_REQUESTED** [review]
Few nits inline.

**Review by hugonicodemos — APPROVED** [review]

**Pull request merged**